httplib.go

```diff
@@ -88,11 +88,11 @@ func replyJSON(w http.ResponseWriter, code int, err error) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(code)
 	var out []byte
-	// trace error can marshal itself,
-	// otherwise capture error message and marshal it explicitly
+	// wrap regular errors in order to achieve unification
+	// and provide structurally consistent responses
 	var obj interface{} = err
 	if _, ok := err.(*TraceErr); !ok {
-		obj = RawTrace{Message: err.Error()}
+		obj = &TraceErr{Err: err}
 	}
 	out, err = json.MarshalIndent(obj, "", "    ")
 	if err != nil {
```

Comment:
Can't comment on the line below, but should it then also be changed to have the structure consistent with these updates? `{"error": {"message": "internal marshal error: %v"}}`

Reply:
I think that this change is in the "nice to have", "optional" domain. The change to remove the stack traces from the API doesn't impact this execution path. So the only benefit of changing this would be to make the API unified. The unification of the API is a side effect of the main goal - minimization of the possible impact to the clients of the trace library due to the removal of the stack traces. I also noticed that there are other places in the Gravity code that still use a simple "message" field to show errors. So the API is not going to be 100% consistent anyways. That is why I decided that there is no need to change that line at this time. 

But on the other hand if we want to change it now it should be ok (at least for Gravity according to my testing).

@a-palchikov  @knisbet  what is your opinion? Do you think that we should change the "internal marshal error" line now?

Reply:
I'd side with @r0mant here to make the change as well. The whole issue with the alternative flat(ter) error message wire format with message on top (`{"message": "..."}`) was another case to support in addition to the settled format so I'm not seeing any compatibility issues, tbh.